server/sonar-server/src/main/java/org/sonar/server/computation/component/ComponentImpl.java

```diff
@@ -83,6 +89,27 @@ public int getRef() {
     return component.getRef();
   }
 
+  public String getUuid() {
+    return uuid;
+  }
+
+  public ComponentImpl setUuid(String uuid) {
+    this.uuid = uuid;
+    return this;
+  }
+
+  public String getKey() {
+    if (key == null) {
+      throw new UnsupportedOperationException(String.format("Component key of ref '%s' has not be fed yet", getRef()));
```

Comment:
if the method throws an exception when the value is not set, this should be documented in javadoc of the method in the `Component` interface